File: app/src/main/java/com/bb/firegooglebooks/animation/Book.java
package com.bb.firegooglebooks.animation;

import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.util.AttributeSet;
import android.view.View;

import androidx.annotation.Nullable;

import com.bb.firegooglebooks.R;

public class Book extends View {

    private Bitmap book;
    private float xPos = 0f;
    private float yPos = 0f;

    private float speed = 10.75f;
    private float xSpeed = speed;
    private float ySpeed = speed;

    private Paint paint = new Paint();

    public Book(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);

        book = BitmapFactory.decodeResource(getResources(), R.drawable.book);
        book = Bitmap.createScaledBitmap(book, (book.getWidth()/2), (book.getHeight()/2), false);
    }

    @Override
    protected void onDraw(Canvas canvas){
        super.onDraw(canvas);

        if(xPos > (getWidth() - book.getWidth()) || xPos < 0)
            xSpeed *= -1;
        if(yPos > (getHeight() - book.getHeight()) || yPos < 0)
            ySpeed *= -1;

        xPos += xSpeed;
        yPos += ySpeed;

        canvas.drawBitmap(book, xPos, yPos, paint);
        invalidate();
    }

}

File: app/src/main/java/com/bb/firegooglebooks/animation/Book.kt
package com.bb.firegooglebooks.animation

import android.content.Context
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.graphics.Canvas
import android.graphics.Paint
import android.util.AttributeSet
import android.view.View
import com.bb.firegooglebooks.R

class Book(context: Context, attrs: AttributeSet) : View(context, attrs) {

    private var book: Bitmap = BitmapFactory.decodeResource(resources, R.drawable.book)

    private var xPos = 0f
    private var yPos = 0f

    private val speed = 10.75f
    private var xSpeed = speed
    private var ySpeed = speed
    private val paint = Paint()

    override fun onDraw(canvas: Canvas) {
        super.onDraw(canvas)
        if (xPos > width - book.width || xPos < 0) xSpeed *= -1f
        if (yPos > height - book.height || yPos < 0) ySpeed *= -1f
        xPos += xSpeed
        yPos += ySpeed
        canvas.drawBitmap(book, xPos, yPos, paint)
        invalidate()
    }

    init {
        book = Bitmap.createScaledBitmap(book, book.width / 2, book.height / 2, false)
    }
}